src/main/java/com/jcs/sbs/auth/Authorization.java

```
@@ -0,0 +1,160 @@
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.jcs.sbs.common.Utils;
+import com.jcs.sbs.model.ProtocolAndHost;
+
+/**
+ * This class contains methods for adding required authorization parameters
+ * before sending the actual request.
+ */
+public class Authorization {
+
+    private String verb;
+    private String accessKey;
+    private String secretKey;
+    private Map<String, String> headers;
+    private String path;
+    private String protocol;
+    private String host;
+    private String port;
+
+    /**
+     * Constructor for initializing data members of this class.
+     * 
+     * @param url
+     *            Request URL on which authorization parameters are to be added
+     * @param verb
+     *            Request method of the API being called
+     * @param accessKey
+     *            Unique Access Key of the JCS account
+     * @param secretKey
+     *            Secret Key of the JCS account
+     * @param headers
+     *            Common headers that is to be added to any API request.
+     */
+    public Authorization(String url, String verb, String accessKey, String secretKey, Map<String, String> headers) {
+        this.verb = verb;
+        this.accessKey = accessKey;
+        this.secretKey = secretKey;
+        this.headers = headers;
+        this.path = "/";
+        ProtocolAndHost protocolAndHost = Utils.getProtocolAndHost(url);
+        if (!Arrays.asList(new String[] { "http", "https" }).contains(protocolAndHost.getProtocol())) {
+            throw new RuntimeException("Unsupported protocol present in given url :" + url);
+        }
+        this.protocol = protocolAndHost.getProtocol();
+        this.host = protocolAndHost.getHost();
+        this.port = "";
+        if (host.indexOf(':') > 0) {
+            String[] parts = host.split(":");
+            this.host = parts[0];
+            this.port = parts[1];
+        }
```

Better to throw an exception if host and port ar not valid. Do you want to consider this ?